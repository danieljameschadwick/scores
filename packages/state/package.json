{
    "name": "@scores/state",
    "version": "0.0.1",
    "type": "module",
    "license": "UNLICENSED",
    "dependencies": {
        "@react-native-async-storage/async-storage": "^1.17.10",
        "@reduxjs/toolkit": "^1.8.3",
        "@scores/types": "^0.0.1",
        "react": "18.2.0",
        "react-devtools": "4.22.0",
        "react-dom": "18.2.0",
        "react-native": "0.69.0",
        "react-redux": "^8.0.2",
        "redux-persist": "^6.0.0",
        "redux-thunk": "^2.4.1"
    }
}
