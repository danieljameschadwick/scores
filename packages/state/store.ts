import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import AsyncStorage from '@react-native-async-storage/async-storage';
import { persistReducer, persistStore } from "redux-persist";
import thunk from "redux-thunk";
import themeReducer from "./reducer/ThemeReducer";

const persistConfig = {
  key: "root",
  storage: AsyncStorage,
};

const persistedReducer = persistReducer(persistConfig, themeReducer);

const store = configureStore({
  reducer: persistedReducer,
  devTools: process.env.NODE_ENV !== "production",
  middleware: [thunk],
});

export const persistor = persistStore(store);

export type AppState = ReturnType<typeof store.getState>;

export type AppDispatch = typeof store.dispatch;

export type AppThunk<ReturnType = void> = ThunkAction<ReturnType,
  AppState,
  unknown,
  Action<string>>;

export default store;
