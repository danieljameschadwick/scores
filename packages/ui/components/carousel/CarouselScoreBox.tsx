import React from "react";
import { View, Text, StyleSheet } from "react-native";
import { getTheme } from "@scores/theme/utils/theme";
import { ScoreBoxRow } from "@scores/ui/components/carousel/scoreBox/ScoreBoxRow";
import { GAME_RESULT } from "@scores/types/enum/GameResult";

interface ScoreInterface {
  abbreviation: string;
  score: number;
  result: GAME_RESULT;
  logo?: string | null;
}

interface Props {
  home: ScoreInterface;
  away: ScoreInterface;
}

export const CarouselScoreBox: React.FC<Props> = ({ home, away }) => {
  const themeStyles = getTheme();

  return (
    <View style={[styles.container, themeStyles.darkContainer]}>
      <Text style={[styles.status, themeStyles.text]}>FT</Text>
      <ScoreBoxRow {...home} />
      <ScoreBoxRow {...away} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: 120,
    display: "flex",
    flexDirection: "column",
    paddingLeft: 12,
    paddingRight: 12,
    paddingTop: 5,
    paddingBottom: 5,
    borderRightWidth: 1,
    backgroundColor: "#FFFFFF",
  },
  status: {
    fontSize: 10,
    fontWeight: "600",
    marginBottom: 3,
  },
});
