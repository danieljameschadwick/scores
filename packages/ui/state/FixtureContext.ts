import { createContext, useContext } from "react";

export const FixtureContext = createContext(null);
export const useFixture = () => useContext(FixtureContext);
