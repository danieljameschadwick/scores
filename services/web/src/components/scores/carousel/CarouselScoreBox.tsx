import { GAME_RESULT } from "@src/enum/GameResult";
import React from "react";
import { View, Text, StyleSheet } from "react-native-web";

const ScoreBoxRow = ({ abbreviation, score, result }) => {
  const teamNameText = [styles.teamName];
  const goalsText = [styles.goalsText];

  if (result === GAME_RESULT.WIN) {
    teamNameText.push(styles.winText);
    goalsText.push(styles.winText);
  }

  if (result === GAME_RESULT.LOSS) {
    teamNameText.push(styles.lossText);
    goalsText.push(styles.lossText);
  }

  return (
    <View style={[styles.row]}>
      <Text style={teamNameText}>{abbreviation}</Text>
      <Text style={goalsText}>{score}</Text>
    </View>
  );
};

export const CarouselScoreBox = ({ home, away }) => {
  return (
    <View style={styles.container}>
      <Text style={[styles.status]}>FT</Text>
      <ScoreBoxRow {...home} />
      <ScoreBoxRow {...away} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: 120,
    display: "flex",
    flexWrap: "wrap",
    flexDirection: "column",
    paddingLeft: 12,
    paddingRight: 12,
    paddingTop: 5,
    paddingBottom: 5,
    borderRightWidth: 1,
    borderRightColor: "rgb(215, 220, 224)",
    borderBottomWidth: 1,
    borderBottomColor: "rgb(215, 220, 224)",
    backgroundColor: "#FFFFFF",
  },
  row: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between",
    marginBottom: 2,
  },
  status: {
    fontSize: 10,
    fontWeight: "600",
    marginBottom: 3,
  },
  teamName: {
    fontSize: 12,
  },
  goalsText: {
    fontSize: 12,
    fontWeight: "600",
  },
  winText: {
    fontWeight: "600",
  },
  lossText: {
    color: "#a5a6a7",
  },
});

const data = {
  fixture: {
    id: 868001,
    referee: "Paul Tierney, England",
    timezone: "UTC",
    date: "2022-09-04T15:30:00+00:00",
    timestamp: 1662305400,
    periods: {
      first: 1662305400,
      second: 1662309000,
    },
    venue: {
      id: 556,
      name: "Old Trafford",
      city: "Manchester",
    },
    status: {
      long: "Match Finished",
      short: "FT",
      elapsed: 90,
    },
  },
  league: {
    id: 39,
    name: "Premier League",
    country: "England",
    logo: "https://media.api-sports.io/football/leagues/39.png",
    flag: "https://media.api-sports.io/flags/gb.svg",
    season: 2022,
    round: "Regular Season - 6",
  },
  teams: {
    home: {
      id: 33,
      name: "Manchester United",
      logo: "https://media.api-sports.io/football/teams/33.png",
      winner: true,
    },
    away: {
      id: 42,
      name: "Arsenal",
      logo: "https://media.api-sports.io/football/teams/42.png",
      winner: false,
    },
  },
  goals: {
    home: 3,
    away: 1,
  },
  score: {
    halftime: {
      home: 1,
      away: 0,
    },
    fulltime: {
      home: 3,
      away: 1,
    },
    extratime: {
      home: null,
      away: null,
    },
    penalty: {
      home: null,
      away: null,
    },
  },
};
