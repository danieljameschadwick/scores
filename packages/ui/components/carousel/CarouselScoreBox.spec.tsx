import React from "react";
import {
  render,
  screen,
  act,
  waitFor,
  fireEvent,
} from "@testing-library/react-native";
import { Provider } from "react-redux";
import Router, { useRouter } from "next/router";
import store, { persistor } from "@scores/state/store";
import { FixtureContext } from "@scores/ui/state/FixtureContext";
import { PersistGate } from "redux-persist/integration/react";
import { CarouselScoreBox } from "./CarouselScoreBox";

const customRender = (fixture) => {
  return render(
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <FixtureContext.Provider value={fixture}>
          <CarouselScoreBox />
        </FixtureContext.Provider>
      </PersistGate>
    </Provider>
  );
};

describe("CarouselScoreBox component", () => {
  it("renders base component", async () => {
    const fixture = {};
    customRender(fixture);

    expect(screen.toJSON()).toMatchSnapshot();
  });

  it("can navigate to fixture", async () => {
    const fixture = {
      id: 12345,
    };
    const { queryByTestId } = customRender(fixture);

    act(() => {
      fireEvent.press(queryByTestId("score-box"));
    });

    await waitFor(() => {
      expect(useRouter.push).toHaveBeenCalled();
    });

    expect(useRouter.push).toHaveBeenCalledWith("/fixture/12345");
    expect(useRouter.push).toHaveBeenCalledWith("/fixture/12345zzzzz");

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
