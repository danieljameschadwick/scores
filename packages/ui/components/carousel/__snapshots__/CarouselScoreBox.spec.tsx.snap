// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselScoreBox component can navigate to fixture 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="score-box"
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRightWidth": 1,
          "display": "flex",
          "flexDirection": "column",
          "paddingBottom": 5,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 5,
          "width": 120,
        },
        Object {
          "backgroundColor": "rgb(232, 234, 237)",
          "borderColor": "rgb(215, 220, 224)",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Array [
            Object {
              "fontSize": 10,
              "fontWeight": "600",
              "marginBottom": 5,
            },
            Object {
              "color": "#000",
            },
            Object {
              "color": "#d00",
            },
          ],
        ]
      }
    >
      LIVE
    </Text>
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 2,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "marginRight": 3,
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": null,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            Array [
              Array [
                Object {
                  "fontSize": 12,
                },
              ],
              Object {
                "color": "#000",
              },
            ]
          }
        />
      </View>
      <Text
        style={
          Array [
            Array [
              Object {
                "fontSize": 12,
                "fontWeight": "600",
              },
            ],
            Object {
              "color": "#000",
            },
          ]
        }
      >
        -
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 2,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "marginRight": 3,
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": null,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            Array [
              Array [
                Object {
                  "fontSize": 12,
                },
              ],
              Object {
                "color": "#000",
              },
            ]
          }
        />
      </View>
      <Text
        style={
          Array [
            Array [
              Object {
                "fontSize": 12,
                "fontWeight": "600",
              },
            ],
            Object {
              "color": "#000",
            },
          ]
        }
      >
        -
      </Text>
    </View>
  </View>
</View>
`;

exports[`CarouselScoreBox component renders base component 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="score-box"
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRightWidth": 1,
          "display": "flex",
          "flexDirection": "column",
          "paddingBottom": 5,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 5,
          "width": 120,
        },
        Object {
          "backgroundColor": "rgb(232, 234, 237)",
          "borderColor": "rgb(215, 220, 224)",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Array [
            Object {
              "fontSize": 10,
              "fontWeight": "600",
              "marginBottom": 5,
            },
            Object {
              "color": "#000",
            },
            Object {
              "color": "#d00",
            },
          ],
        ]
      }
    >
      LIVE
    </Text>
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 2,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "marginRight": 3,
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": null,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            Array [
              Array [
                Object {
                  "fontSize": 12,
                },
              ],
              Object {
                "color": "#000",
              },
            ]
          }
        />
      </View>
      <Text
        style={
          Array [
            Array [
              Object {
                "fontSize": 12,
                "fontWeight": "600",
              },
            ],
            Object {
              "color": "#000",
            },
          ]
        }
      >
        -
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 2,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "marginRight": 3,
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": null,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            Array [
              Array [
                Object {
                  "fontSize": 12,
                },
              ],
              Object {
                "color": "#000",
              },
            ]
          }
        />
      </View>
      <Text
        style={
          Array [
            Array [
              Object {
                "fontSize": 12,
                "fontWeight": "600",
              },
            ],
            Object {
              "color": "#000",
            },
          ]
        }
      >
        -
      </Text>
    </View>
  </View>
</View>
`;
